3rdparty/mshadow/mshadow/bfloat.h

```diff
@@ -0,0 +1,167 @@
+/*!
```

Comment:
@szha Do we need Apache license header for this new file?

Reply:
yes

Reply:
done  thanks